phpstan-baseline.neon

```diff
@@ -345,6 +300,36 @@ parameters:
 			count: 1
 			path: tests/Sabre/Xml/Deserializer/FunctionCallerTest.php
 
+		-
+			message: "#^Parameter \\#2 \\$func of function Sabre\\\\Xml\\\\Deserializer\\\\functionCaller expects callable\\(\\)\\: Sabre\\\\Xml\\\\Deserializer\\\\Address, 'Sabre\\\\\\\\Xml…' given\\.$#"
```

Comment:
these tests use some pretty exotic notations for declaring a callable in https://github.com/sabre-io/xml/blob/584d601a7bdec8af7e5de0b18b9b4398249a7bde/tests/Sabre/Xml/Deserializer/FunctionCallerTest.php#L100-L105

I am fine with reporting these as errors, wdyt?

Reply:
Actually `functionCaller` is happy to accept a string that is a class, and anything that is a callable - regardless of the parameters of the callable.

The doc line `@param callable():R $func` in `functionCaller` seems to force PHPstan to only allow passing a callable that is a function with no parameters.

I removed that doc line in commit https://github.com/sabre-io/xml/pull/216/commits/cc4ce0635fb2a84d6ef3900022face9e99114a55 and got rid of these lines of "expected messages" and it passes.

In the tests, locally, I tried passing some string that is not a class/method that exists and that makes phpstan complain. So it is checking that a real callable is being passed.

Maybe cherry-pick that commit?

Reply:
dropping the phpdoc will remove the tooling/type coverage. but I wasn't able to figure out a way, to make phpstan analyse this code properly. so leaving it out for now, as you suggested.